### PR TITLE
Add Glean overview article

### PR DESCRIPTION
**Pull request by georgf**
I'm adding a first pass on a Glean overview article, so we have a single place to point to.
This doesn't have to be the final version, we can iterate on it.

The goal is to have a landing page for Engineers from other teams, Engineering Managers, Product Managers, etc. This means giving them a basic overview of what Glean is and where to find more information.

### CODE REVIEW

File: src/concepts/glean.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Glean - mobile product analytics
+
+For Mozilla, getting reliable data from our products is critical to inform our decision making. Glean is our new product analytics & Telemetry solution that provides that data for our mobile products.
```
Comment (Dexterp37) [inline src/concepts/glean.md]: nit: is telemetry supposed to start with a lowercase letter here?

File: src/concepts/glean.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+*   [What does it offer](#what-does-it-offer)
+*   [How to use Glean](#how-to-use-glean)
+*   [Contact](#contact)
+*   [References](#references)
+
+# Overview
+
+Glean consists of different pieces:
+
+*   Product-side tools - [The Glean SDK](https://github.com/mozilla-mobile/android-components/blob/master/components/service/glean/README.md) is what products integrate and record data into.
```
Comment (Dexterp37) [inline src/concepts/glean.md]: nit: lower case `T` in `The Glean SDK`

File: src/concepts/glean.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+*   [Integrate the Glean SDK / library](https://github.com/mozilla-mobile/android-components/blob/master/components/service/glean/README.md) into your product.
+*   [File a data engineering bug](https://bugzilla.mozilla.org/enter_bug.cgi?product=Data%20Platform%20and%20Tools&component=General&short_desc=Glean:%20Enable%20application%20id%20org.mozilla.myProduct) to enable your products application id.
+*   [Use Redash](https://sql.telemetry.mozilla.org/) to write SQL queries & build dashboards using your products datasets, e.g.:
+    *   `_org_mozilla_fenix_baseline_parquet`
+    *   `_org_mozilla_fenix_events_parquet`
+    *   `_org_mozilla_fenix_metrics_parquet`
+*   _(Work in progress)_ Use events and [Amplitude](https://sso.mozilla.com/amplitude) for product analytics.
+*   [Use Databricks](https://sso.mozilla.com/databricks) for deep-dive analysis.
+*   Use the Glean debug ping viewer for QA & development.
```
Comment (Dexterp37) [inline src/concepts/glean.md]: Should we link [this document](https://docs.google.com/document/d/1LdXgXxOTDQQqD7_yGYj9TE80JQ3Wq2nx5vrgn7oDqt8/edit#) here?

File: src/concepts/glean.md (hunk 1 above)
Comment (mdboom) [reply src/concepts/glean.md]: Also, with desktop on the roadmap, do we want to use the word "mobile" here?  (Or maybe that's too soon...)

File: src/concepts/glean.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+A baseline of analysis is important for all our mobile application, from counting active users to retention and session times. This is supported out-of-the-box by the library and work consistently across our mobile products.
+
+**No custom code is required for adding new metrics to a product.**
+To make engineers more productive, the SDK keeps the amount of instrumentation code required for metrics as small as possible. Engineers only need to specify what they want to instrument, with which semantics and then record the data using the Glean SDK. The SDK takes care of storing & sending that data reliably.
+
+**Following [lean data practices](https://leandatapractices.com/) through SDK design choices.**
+It's easy to limit data to what's necessary and documentation can be generated easily, aiding both transparency & understanding for analysis.
+
+**Better data tooling integration due to standardized data types & registering them in machine-readable files.**
+When having collected data described in machine-readable files, our various data tools can read them and support metrics automatically, without manual work.
```
Comment (mdboom) [inline src/concepts/glean.md]:
```suggestion
By having collected data described in machine-readable files, our various data tools can read them and support metrics automatically, without manual work.
```

File: src/concepts/glean.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+**Due to common high-level concepts for metrics, APIs & data tools can better match the use-cases.**
+To make the choice easier for which metric type to use, we are introducing higher-level data types that offer clear and understandable semantics - when you want to count something, you use the _"count"_ type. This also gives us opportunities to offer better tooling for the data, both on the client and for data tooling.
+
+**Basic semantics on how the data is collected are clearly defined by the library.**
+To make it easier to understand the general semantics of our data, the Telemetry library will define and document when which kind of data will get sent. This gives data analysis common basic semantics.
```
Comment (mdboom) [inline src/concepts/glean.md]: Maybe `Telemetry` -> `Glean`?

File: src/concepts/glean.md (hunk 1 above)
Comment (georgf) [reply src/concepts/glean.md]: Currently Glean provides mobile product analytics only, so i think for now it's good to call that out.
We can say somewhere that Desktop is considered/planned, but we couldn't even put a firm date on that yet.

File: src/concepts/glean.md (hunk 3 above)
Comment (georgf) [reply src/concepts/glean.md]: Good point!
We should probably move that document into the firefox-data-docs too once @akkomar isn't concerned about making the URL public anymore.

File: src/concepts/glean.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+**Following [lean data practices](https://leandatapractices.com/) through SDK design choices.**
+It's easy to limit data to what's necessary and documentation can be generated easily, aiding both transparency & understanding for analysis.
+
+**Better data tooling integration due to standardized data types & registering them in machine-readable files.**
+When having collected data described in machine-readable files, our various data tools can read them and support metrics automatically, without manual work.
+
+**Due to common high-level concepts for metrics, APIs & data tools can better match the use-cases.**
+To make the choice easier for which metric type to use, we are introducing higher-level data types that offer clear and understandable semantics - when you want to count something, you use the _"count"_ type. This also gives us opportunities to offer better tooling for the data, both on the client and for data tooling.
```
Comment (fbertsch) [inline src/concepts/glean.md]:
```suggestion
To make the choice easier for which metric type to use, we are introducing higher-level data types that offer clear and understandable semantics - for example, when you want to count something, you use the _"count"_ type. This also gives us opportunities to offer better tooling for the data, both on the client and for data tooling.
```

File: src/concepts/glean.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
+# Glean - mobile product analytics
+
+For Mozilla, getting reliable data from our products is critical to inform our decision making. Glean is our new product analytics & Telemetry solution that provides that data for our mobile products.
+It aims to be easy to integrate, reliable and transparent by providing an SDK and integrated tools.
+
+It currently supports Android products, while iOS support is planned.
+Note that this is different from Telemetry for Firefox Desktop ([1](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/index.html), [2](https://docs.telemetry.mozilla.org/concepts/choosing_a_dataset.html)), although it provides similar capabilities.
```
Comment (badboy) [inline src/concepts/glean.md]: Can we give these links proper names? `1` and `2` are bad click targets.

File: src/concepts/glean.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+Contents:
```
Comment (badboy) [inline src/concepts/glean.md]: This article is rather short. Do we need a ToC?

In any case `<!-- toc -->` should give that automatically.

File: src/concepts/glean.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+Glean is designed to support typical product analytics use-cases and encourage best practices by requiring clearly defined metrics through the following:
+
+**Basic product analytics are collected out-of-the-box in a standardized way.**
+A baseline of analysis is important for all our mobile application, from counting active users to retention and session times. This is supported out-of-the-box by the library and work consistently across our mobile products.
+
+**No custom code is required for adding new metrics to a product.**
+To make engineers more productive, the SDK keeps the amount of instrumentation code required for metrics as small as possible. Engineers only need to specify what they want to instrument, with which semantics and then record the data using the Glean SDK. The SDK takes care of storing & sending that data reliably.
+
+**Following [lean data practices](https://leandatapractices.com/) through SDK design choices.**
+It's easy to limit data to what's necessary and documentation can be generated easily, aiding both transparency & understanding for analysis.
```
Comment (badboy) [inline src/concepts/glean.md]:
```suggestion
It's easy to limit data collection to what's necessary and documentation can be generated easily, aiding both transparency & understanding for analysis.
```

File: src/concepts/glean.md (hunk 5 above)
Comment (badboy) [inline src/concepts/glean.md]: Telemetry -> Glean?

"when which kind of data" -> "when and what/which kind of data" ?

Comment (georgf) [reply src/concepts/glean.md]: That second suggestion sounds harder to understand/parse to me, dropping it.

**Comment by georgf** [comment]
The remaining CI error is unrelated:
> FILE: src/datasets/shield.md
> [✖] https://firefox-test-tube.herokuapp.com/ → Status: 404

**Pull request merged**